word_cloud_generator/word_cloud_generator.py

```diff
@@ -0,0 +1,254 @@
+    )
+    args = parser.parse_args(inp_args)
+    return args
+
+
+def get_response_json(auth, headers, url):
+    """
+    Function to send a get request to a url and return the response as json
+    :param auth: A HTTPBasicAuth object for authentication (HTTPBasicAuth)
+    :param headers: A request Header (dict)
+    :param url: The URL to send the request (string)
+    :returns: A dictionary of JSON values
+    """
+    session = requests.session()
+    session.headers = headers
+    session.auth = auth
+
+    attempts = 5
+
+    while attempts > 0:
+        response = session.get(url, timeout=5)
+        if (
+            response.content != b'{"status":"RUNNING"}'
+            and response.content != b'{"status":"ENQUEUED"}'
+        ):
+            break
+        else:
+            sleep(1)
+            attempts = attempts - 1
+
+    if attempts == 0:
+        raise requests.exceptions.Timeout(
+            "Get request status not completed before timeout"
+        )
+
+    return json.loads(response.text)
+
+
+def get_issues_contents_after_time(auth, headers, host, issue_filter):
+    """
+    Function to get the contents of through issues using a loop, as only 50 can be checked at a time
+    :param issue_filter: Dict of filters to check the issues against (dict)
+    :param auth: A HTTPBasicAuth object for authentication (HTTPBasicAuth)
+    :param headers: A request Header (dict)
+    :param host: The host used to create the URL to send the request (string)
+    :returns: A list with the contents of all valid issues
+    """
+    issues_contents = []
+    return_amount = 50
+    issues_amount = 0
+    while return_amount == 50:
+        url = f"{host}/rest/servicedeskapi/servicedesk/6/queue/182/issue?start={issues_amount}"
+
+        json_load = get_response_json(auth, headers, url)
+
+        issues = json_load.get("values")
+
+        for issue in issues:
+            issue_date = datetime.strptime(
+                issue.get("fields").get("created")[:10], "%Y-%m-%d"
+            )
+            if issue_date < datetime.strptime(issue_filter.get("end_date"), "%Y-%m-%d"):
+                return issues_contents
+            if filter_issue(issue, issue_filter, issue_date):
+                issue_contents = issue.get("fields").get("summary")
+                if issue_contents:
+                    issues_contents.append(issue_contents)
+
+        return_amount = json_load.get("size")
+        issues_amount = issues_amount + return_amount
+
+    return issues_contents
+
+
+def filter_issue(issue, issue_filter, issue_date):
+    """
+    Function to check if an issue passes the set filters
+    :param issue: A dict of an issues contents (dict)
+    :param issue_filter: Dict of filters to check the issues against (dict)
+    :param issue_date: The date that the issue was created (string)
+    :returns: If the issue passes the filters
+    """
+    if issue.get("fields").get("assignee"):
```

Comment:
this is fine but is there a reason you are using `get` instead of issue["fields"]["assignee"]?

Using `get` implies that you want to include a default if the key is not found - which you might want to do

if you want to return false if the key isn't found - you could rewrite it like so:

```
def filter_issue(issue, issue_filter, issue_date):
    """
    Function to check if an issue passes the set filters
    :param issue: A dict of an issues contents (dict)
    :param issue_filter: Dict of filters to check the issues against (dict)
    :param issue_date: The date that the issue was created (string)
    :returns: If the issue passes the filters
    """
    
    fields = issue.get("fields", None)
    if not fields:
        return False

    assignee = fields.get("assignee", None)
    if not assignee:
        return False

    issue_assigned = assignee.get("displayName", None)
    assign_check = issue_filter.get("assigned", None)
    if not issue_assigned or not assign_check or issue_assigned != assign_check:
        return False

    if issue_date > datetime.strptime(issue_filter.get("start_date"), "%Y-%m-%d"):
        return False
    return True
```